gui/quantum.lua

```diff
@@ -11,262 +11,212 @@ local quickfort = reqscript('quickfort')
-                key='CUSTOM_R',
-                label='Allow refuse/corpses',
-                initial_option=false},
-            widgets.TooltipLabel{
-                text_to_wrap='Note that enabling refuse will cause clothes' ..
+            text_pen=COLOR_GREY
+		},
+		widgets.HotkeyLabel{
+            key='CUSTOM_Q',
+            label='Place QSP under cursor',
+            on_activate=self:callback('commit')
+		},
+        widgets.Panel{autoarrange_subviews=true, 
+			frame = { w=75, h=5 },
+			resizable = true,
+			subviews={
+				widgets.EditField{
+					view_id='name',
+					key='CUSTOM_N',
+					on_char=self:callback('on_name_char'),
+					text=''
+				},
+				widgets.TooltipLabel{
+					text_to_wrap='Give the quantum stockpile a custom name.',
+					show_tooltip=true
+				}
+			}
+		},
+        widgets.Panel{
+			autoarrange_subviews=true, 
+			frame = { w=75, h=5 },
+			resizable = true,
+			subviews={
+				widgets.CycleHotkeyLabel{
+					view_id='dir',
+					key='CUSTOM_D',
+					options={
+						{label='North', value={y=-1}},
+                        {label='South', value={y=1}},
+                        {label='East', value={x=1}},
+                        {label='West', value={x=-1}}
+					}
+				},
+				widgets.TooltipLabel{
+					text_to_wrap='Set the dump direction of the quantum stop.',
+					show_tooltip=true
+				}
+			}
+		},
+        widgets.Panel{
+			autoarrange_subviews=true, 
+			frame = { w=75, h=5 },
+			resizable = true,
+			subviews={
+				widgets.ToggleHotkeyLabel{
+					view_id='refuse',
+					key='CUSTOM_R',
+					label='Allow refuse/corpses',
+					initial_option=false
+				},
+				widgets.TooltipLabel{
+					text_to_wrap='Note that enabling refuse will cause clothes' ..
                     ' and armor in this stockpile to quickly rot away.',
-                show_tooltip=true}}},
+					show_tooltip=true
+				}
+			}
+		},
         widgets.WrappedLabel{
-            text_to_wrap=('%d minecart%s available: %s will be %s'):format(
+            text_to_wrap=('%d minecart%s available: %s %s'):format(
                 cart_count, cart_count == 1 and '' or 's',
                 cart_count == 1 and 'it' or 'one',
-                cart_count > 0 and 'automatically assigned to the quantum route'
-                    or 'ordered via the manager for you to assign later')},
+                cart_count > 0 and 'will be automatically assigned to the quantum route'
+                    or 'will need to be ordered for you to assign later')
+		},
         widgets.HotkeyLabel{
             key='LEAVESCREEN',
             label=self:callback('get_back_text'),
-            on_activate=self:callback('on_back')}
+            on_activate=self:callback('on_back')
+		}
     }
 
     self:addviews{main_panel}
 end
 
+-- UI Functions
 function QuantumUI:get_help_text()
-    if not self.feeder then
+    if not dfhack.gui.getSelectedStockpile() then
         return 'Please select the feeder stockpile with the cursor or mouse.'
```

Comment:
there isn't really a cursor option for this anymore. directions should be to use the mouse